packages/mc-scripts/src/utils/get-config-diff.js

```diff
@@ -0,0 +1,246 @@
+const { red, green } = require('chalk');
+
+const notNestedKeySet = new Set(['name', 'description', 'url', 'icon']);
+
+const t = (tabs) => '\t'.repeat(tabs);
+
+const getStringDiff = (oldStr, newStr, key, tabs = 0) => {
+  if (!oldStr && newStr) {
+    return `${t(tabs)}${key} added: ${green(newStr)}`;
+  } else if (oldStr && !newStr) {
+    return `${t(tabs)}${key} removed: ${red(oldStr)}`;
+  } else if (oldStr && newStr && oldStr !== newStr) {
+    return `${t(tabs)}${key} changed: ${red(oldStr)} => ${green(newStr)}`;
+  }
+};
+
+const getPermissionsDiff = (oldPermissions, newPermissions) => {
+  const permissionDiff = ['permissions changed'];
+  const mappedOldPermission = {};
+
+  oldPermissions.forEach(({ name, oAuthScopes }) => {
+    mappedOldPermission[name] = oAuthScopes;
+  });
+
+  newPermissions.forEach((newPermission) => {
+    const currDiff = [`\t${newPermission.name} changed`];
+    // if the permission name is not in the old config, it means it is a new addition.
+    if (!mappedOldPermission[newPermission.name]) {
+      permissionDiff.push(`\t"${green(newPermission.name)}" was added`);
+    }
+
+    // if permission name is in the old config, now we check if there was a change
+    else {
+      currDiff.push(
+        getArrayDiff(
+          mappedOldPermission[newPermission.name],
+          newPermission.oAuthScopes,
+          2
+        )
+      );
+      delete mappedOldPermission[newPermission.name];
+    }
+
+    currDiff.filter(Boolean).length > 1 &&
+      permissionDiff.push(currDiff.join('\n'));
+  });
+
+  // if there are old permissions left, it means they were deleted in the new Permissions.
+  Object.keys(mappedOldPermission).forEach((oldPermissionName) => {
+    permissionDiff.push(`\t"${red(oldPermissionName)}" was removed`);
+  });
+
+  if (permissionDiff.length > 1) return permissionDiff.join('\n');
+};
+
+const getArrayDiff = (oldArray, newArray, tabs = 0) => {
+  const oldArraySet = new Set(oldArray);
+  const arrayDiff = [];
+
+  newArray?.forEach((item) => {
+    if (oldArraySet.has(item)) {
+      oldArraySet.delete(item);
+    } else {
+      arrayDiff.push(`${t(tabs)}Item with "${green(item)}" added`);
```

Comment:
Nit: can't we use the `getStringDiff` function?

```suggestion
      arrayDiff.push(getStringDiff(null, item, 'Item with', indentLevel));
```

Reply:
The wording for getStringDiff is different. That's why I didn't use it.